lib/markdown-service.js

```diff
@@ -66,8 +66,11 @@ class MarkdownService {
             data = data.replace(/<strong>|<\/strong>/g, '**');
 
             // add escape for numbered lists
-            // https://regex101.com/r/nB8bV3/1
-            data = data.replace(/(^\d{1,2})(. )/, '$1\\\\. ');
+            // https://regex101.com/r/SO5haw/1
+            if (data.match(/^\d{1,4}\./)) {
+                // https://regex101.com/r/nB8bV3/4
+                data = data.replace(/(^\d{1,4})(.)/, '$1\\$2');
```

Comment:
Now that you mention it, yes, it would be better to do away with the conditional entirely since the body of the conditional is only one line.